rfc/networked-dot-storage.md

```diff
@@ -0,0 +1,174 @@
+
+But, yes, we're going to try to build this. So given that, what missing pieces SHOULD be built in the next six months? 
+
+**This is a loosely prioritized of inituitions. It's not final and it's open to discussions.**
+
+*Btw, we're not alone here -- because we're ultimately building a network, a critical one to the overal PLVerse vision, we can draw from other teams, and eventually, some of the required components may not be built by the W3S*
+
+### Other implementations 
+
+The first step on the path for W3S is indeed to build other implementations of w3up. Local.storage is our first experiment. It would be nice to produce a cloudflare only implementation. The architecture proposed in [Core Node Architecture](https://w3sat.notion.site/Filecoin-SPs-in-the-W3S-Network-16a9b66e7ef3425bbee390c6481bd752?pvs=4) is a kind of aspirational target -- an ultraportable implementation that could run in most server environments, be ultra-sandboxed, provide great update mechanisms, and support rapid iteration.
+
+We should constinue to explore these options.
+
+### Router.storage
+
+We need an implementation of W3up that simply routes to other instances of W3Up. Building this implies several tasks:
+- deciding who gets each piece of content
+- how to manage UCAN delegations across relays
+
+And eventually:
+- how to allow nodes to join and how to decide when to kick them out
+
+But first, we should probably just hardcode a list of nodes.
+
+*** But isn't this a point of centralization if everyone uses it? ***
+
+Absolutely. It's important that key coordination functions in the network like router.storage eventually operate through distributed consensus and produce verifiable results.
+
+### Read Pipeline that finds content from many write pipelines
+
+A retrieval service (i.e. Freeway) should not be tied to a specific instance of w3up. This implies some form of global content discovery. [W3 IPNI Indexing RFC](https://github.com/web3-storage/RFC/blob/main/rfc/ipni-w3c.md) is a major step in this direction. 
+
+Eventually we may need some kind of regional caching.
+
+### Repair service for both W3UP and Filecoin
+
+Repair service means a service that creates permaneant reliable storage from a series of time-limited storage deals with providers who might fail.
+
+It's a vital missing component today on Filecoin AND a missing component of W3S as soon as their is more than one instance. We've already encountered this problem indirectly as we've thought about deal renewals and the reality that deal terms on Filecoin do not coincide in time with users joining and leaving W3UP.
+
+It needs to exist for Filecoin AND it needs to exist for W3S.
+
+*** But isn't this a point of centralization if everyone uses it? ***
+
+Absolutely. It's important that key coordination functions in the network like a repair service eventually operate through distributed consensus and produce verifiable results.
```

Comment:
Similar question as above re: distributed consensus - I don't think this is a point of centralization if there are other nodes on the network that can perform this service for people. Maybe the blockchain provides a coordination point so people can easily hand off to other "repair services" - some level of that makes sense to me, though I still worry about it being a single point of failure in that process!

Reply:
yea perhaps repair service isn't the right thing for a blockchain, cause you generally might have some level of choice there.

here's the way I look at it:
- single service = single point of failure
- multiple services = no single point of failure, lots of choices, but no coordination, which often reduces user value
- blockchain = acts like single service, but is actually resilient no single point of failure system, and produces more trustworthy output (usually) -- there are times you want this, times you really just want to give people choices. 

Also, personally I do think some of the large L1s can be considered pretty reliable at this point. (though utility is less proven)

Reply:
How about alternative view of the repair service. Instead of having a single blockchain operated system, allow users to run repair service anywhere they like, IPVM, AWS, on user device in their closet or whatever else (in fact I think we could have interesting offering here). In a sense repair service become an agent hired by user which acts on user behalf.

Point I'm trying to make here is that coordination is both difficult and costly and I'm not sure it provides any real benefit to the user that needing assurance of service been operational. Also note that if repair service is just a code you can run it can probably also run as smart contract if users would want to do that.

Overall I'm for plurality of choice, as opposed to assumption that there is a universal solution that fits everyones needs.

Reply:
Yea, I get that choice is your go to.

Personally, I feel that the vast majority of every day users are not going to want to get down and dirty on these details, which I think are pretty low level. Which is why I lean towards sensible defaults first. But I'm not opposed to choice.

To be clear though I think that:
1. Repair service is absolutely something there can be many of.
2. I really misspoke, the default repair service SHOULD simply be a smart contract.